### PR TITLE
Require and execute suite.yaml files

### LINKED ISSUE #164: Require existence of suite.yaml and run gator test as part of presubmit
We should require that new submissions have a suite.yaml living next to the template.

We should also run gator test as part of presubmit.

@willbeason

### PR DESCRIPTION
**Pull request by julianKatz**
This PR adds github actions to:

1. Require a `suite.yaml` file alongside every `template.yaml` file
2. Execute the `suite.yaml` files as a Github Action

Fixes #164

### CODE REVIEW

File: .github/workflows/workflow.yaml [hunk 1]
```diff
@@ -53,3 +53,18 @@ jobs:
           name: logs
           path: |
             logs-*.json
+  require_suites:
```
Comment (willbeason) [inline .github/workflows/workflow.yaml:56]: Any reason to have these as separate actions instead of one?

**Review by willbeason — APPROVED** [review]

File: .github/workflows/workflow.yaml (hunk 1 above)
Comment (julianKatz) [reply .github/workflows/workflow.yaml:56]: This way I can know what the problem is without having to look at the log output directly.  I'd just see the failed job and know what to go debug.

They also run in parallel, though the speedup likely isn't meaningful.

**Review by maxsmythe — APPROVED** [review]
LGTM

Thanks for doing this!

File: .github/workflows/workflow.yaml (hunk 1 above)
Comment (willbeason) [reply .github/workflows/workflow.yaml:56]: Sounds good to me!

**Review by ritazh — APPROVED** [review]
lgtm

pending rebase and address merge conflict

**Pull request merged**